Source/WebCore/html/HTMLInputElement.cpp

```diff
@@ -1007,15 +1011,15 @@ bool HTMLInputElement::isSuccessfulSubmitButton() const
 {
     // HTML spec says that buttons must have names to be considered successful.
     // However, other browsers do not impose this constraint. So we do not.
-    return m_inputType->canBeSuccessfulSubmitButton();
+    return protectedInputType()->canBeSuccessfulSubmitButton();
 }
 
 bool HTMLInputElement::matchesDefaultPseudoClass() const
 {
     ASSERT(m_inputType);
-    if (m_inputType->canBeSuccessfulSubmitButton())
+    if (protectedInputType()->canBeSuccessfulSubmitButton())
```

Comment:
We should use a local variable to avoid repeated calls to `protectedInputType()`